tests/unit_tests/test_unit.py

```
@@ -851,6 +861,61 @@ def test_get_weights_from_url(monkeypatch):
+        assert cache_file_path.read_bytes() == MockResponseGet.file_content
+        assert successful_get.request_counter == 1
+
+
+def test_download_weights_discards_temp_file_on_keyboard_interrupt(
+    monkeypatch, tmp_path
+):
+    download_path = tmp_path / "model_weights.ckpt"
+    mock_get = MockResponseGet()
+    monkeypatch.setattr(requests, "get", mock_get)
+
+    def fail_copyfileobj(source, destination):
```

Ruff (0.16.3)

[warning] Missing return type annotation for private function `fail_copyfileobj`

Add return type annotation: `NoReturn`

(ANN202)